homeassistant/components/tuya/light.py

```diff
@@ -402,7 +401,10 @@ def __init__(
             )
 
         # Update internals based on found brightness dpcode
-        if self._brightness_dpcode:
+        if (
+            self._brightness_dpcode
+            and self._brightness_dpcode in device.status_range
```

Comment:
I'm not sure why this check is done here?
Additionally, the functionality is already checked above, before `_brightness_dpcode` is set.

Could you elaborate on this?

Reply:
`device.status_range[self._brightness_dpcode].values` triggered an error:
```
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 249, in _async_setup_platform
    await asyncio.shield(task)
  File "/config/custom_components/tuyaFixed/light.py", line 316, in async_setup_entry
    async_discover_device([*hass_data.device_manager.device_map])
  File "/config/custom_components/tuyaFixed/light.py", line 309, in async_discover_device
    TuyaLightEntity(
  File "/config/custom_components/tuyaFixed/light.py", line 408, in __init__
    device.status_range[self._brightness_dpcode].values
KeyError: <DPCode.BRIGHT_CONTROLLER: 'bright_controller'>
```

Reply:
That indicates there is an API issue, as the device has the key set in the function. Have you contacted Tuya about it?

Reply:
The lack of protection in this integration is a bug in this integration and should be fixed in this integration regardless of what information you get from the API.
I didn't contact Tuya, I wouldn't even know what to ask because I don't know the API. I simply fixed the bug.

Reply:
> The lack of protection in this integration is a bug in this integration

That I could agree on, however, it is a bug upstream that the functionality is exposed but the range isn't provided.

Reply:
@frenck do you have a better solution for #62838 than mine?

Reply:
We can't fix a third-party API. See my answers above

Reply:
Let's not fix someone else's API, let's teach our integration what to do when the API goes wrong.

Reply:
> I don't want to fix someone else's API, I want to teach our integration what to do when the API goes wrong.

And we don't want to have device-specific exceptions in our code. This is a clear issue with Tuya, which they are known to fix. Please report it to Tuya.

Reply:
this fix is not device-specific. It won't change the way integration works for the devices for which it works at all.

Reply:
This change is not correct and is an issue with the device in the API of Tuya. Please contact them about this issue so they can fix it.

The suggested solution is not correct.

We are not interested in building workarounds.